From e006abae25ee5810b099dc95393e9fd6e5d26627 Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Wed, 25 Jun 2025 11:49:47 +0200
Subject: [PATCH] a

Signed-off-by: Sefa Eyeoglu <contact@scrumplex.net>
---
 src/basic/terminal-util.c | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/src/basic/terminal-util.c b/src/basic/terminal-util.c
index 409f2abb45..140d17a5a8 100644
--- a/src/basic/terminal-util.c
+++ b/src/basic/terminal-util.c
@@ -122,18 +122,18 @@ int read_one_char(FILE *f, char *ret, usec_t t, bool *need_nl) {
                 new_termios.c_cc[VMIN] = 1;
                 new_termios.c_cc[VTIME] = 0;
 
-                if (tcsetattr(fd, TCSADRAIN, &new_termios) >= 0) {
+                if (tcsetattr(fd, TCSANOW, &new_termios) >= 0) {
                         char c;
 
                         if (t != USEC_INFINITY) {
                                 if (fd_wait_for_event(fd, POLLIN, t) <= 0) {
-                                        (void) tcsetattr(fd, TCSADRAIN, &old_termios);
+                                        (void) tcsetattr(fd, TCSANOW, &old_termios);
                                         return -ETIMEDOUT;
                                 }
                         }
 
                         r = safe_fgetc(f, &c);
-                        (void) tcsetattr(fd, TCSADRAIN, &old_termios);
+                        (void) tcsetattr(fd, TCSANOW, &old_termios);
                         if (r < 0)
                                 return r;
                         if (r == 0)
@@ -1915,7 +1915,7 @@ int get_default_background_color(double *ret_red, double *ret_green, double *ret
         struct termios new_termios = old_termios;
         termios_disable_echo(&new_termios);
 
-        if (tcsetattr(STDIN_FILENO, TCSADRAIN, &new_termios) < 0)
+        if (tcsetattr(STDIN_FILENO, TCSANOW, &new_termios) < 0)
                 return -errno;
 
         r = loop_write(STDOUT_FILENO, ANSI_OSC "11;?" ANSI_ST, SIZE_MAX);
@@ -1979,7 +1979,7 @@ int get_default_background_color(double *ret_red, double *ret_green, double *ret
         }
 
 finish:
-        RET_GATHER(r, RET_NERRNO(tcsetattr(STDIN_FILENO, TCSADRAIN, &old_termios)));
+        RET_GATHER(r, RET_NERRNO(tcsetattr(STDIN_FILENO, TCSANOW, &old_termios)));
         return r;
 }
 
@@ -2106,7 +2106,7 @@ int terminal_get_size_by_dsr(
         struct termios new_termios = old_termios;
         termios_disable_echo(&new_termios);
 
-        if (tcsetattr(input_fd, TCSADRAIN, &new_termios) < 0)
+        if (tcsetattr(input_fd, TCSANOW, &new_termios) < 0)
                 return log_debug_errno(errno, "Failed to to set new terminal settings: %m");
 
         unsigned saved_row = 0, saved_column = 0;
@@ -2209,7 +2209,7 @@ finish:
         if (saved_row > 0 && saved_column > 0)
                 RET_GATHER(r, terminal_set_cursor_position(output_fd, saved_row, saved_column));
 
-        RET_GATHER(r, RET_NERRNO(tcsetattr(input_fd, TCSADRAIN, &old_termios)));
+        RET_GATHER(r, RET_NERRNO(tcsetattr(input_fd, TCSANOW, &old_termios)));
         return r;
 }
 
-- 
2.49.0

