diff --git a/src/utils/realtime.rs b/src/utils/realtime.rs
index 9ee8b51f..117d346b 100644
--- a/src/utils/realtime.rs
+++ b/src/utils/realtime.rs
@@ -21,6 +21,15 @@ pub fn set_rt_scheduling() {
         0 => (),
         _ => warn!("unknown failure setting real-time policy: {res}"),
     }
+
+    let res = unsafe {
+        libc::prctl(libc::PR_CAP_AMBIENT, libc::PR_CAP_AMBIENT_CLEAR_ALL, 0, 0, 0)
+    };
+
+    match res {
+        0 => (),
+        _ => warn!("unknown failure dropping all ambient capabilities: {res}"),
+    }
 }
 
 #[cfg(not(target_os = "linux"))]
